### PR TITLE
Fix Hiking trail(s) get(s) split #16058

### PR DESCRIPTION
**Pull request by Dima-1**

### CODE REVIEW

**Comment by vshcherb** [comment]
Queue is part of algorithm connection but loadRouteSegmentTile is generic method that should return generic list.

**Review by vshcherb — CHANGES_REQUESTED** [review]

File: OsmAnd-java/src/main/java/net/osmand/router/network/NetworkRouteSelector.java [hunk 1]
```diff
@@ -549,6 +550,28 @@ private void loadData(NetworkRouteSegment segment, RouteKey rkey, List<NetworkRo
 		}
 	}
 
+	private void addSegmentsBiggerThanOneTile(Map<RouteKey, List<NetworkRouteSegment>> tiles,
+	                                          long tileId, TLongArrayList queue) {
+		for (Map.Entry<RouteKey, List<NetworkRouteSegment>> entry : tiles.entrySet()) {
```
Comment (ivanPyrohivskyi) [inline OsmAnd-java/src/main/java/net/osmand/router/network/NetworkRouteSelector.java:555]: if queue is null just return

Comment (Dima-1) [reply OsmAnd-java/src/main/java/net/osmand/router/network/NetworkRouteSelector.java:555]: queue is not null by default

File: OsmAnd-java/src/main/java/net/osmand/router/network/NetworkRouteSelector.java [hunk 2]
```diff
@@ -549,6 +550,28 @@ private void loadData(NetworkRouteSegment segment, RouteKey rkey, List<NetworkRo
...
+			for (NetworkRouteSegment segment : entry.getValue()) {
+				RouteDataObject rdo = segment.robj;
+				if (rdo == null) {
+					continue;
+				}
+				long startPointTileId = NetworkRouteContext.getTileId(rdo.getPoint31XTile(segment.start), rdo.getPoint31YTile(segment.start));
+				long endPointTileId = NetworkRouteContext.getTileId(rdo.getPoint31XTile(segment.end), rdo.getPoint31YTile(segment.end));
+				if (queue != null) {
+					if (tileId != startPointTileId) {
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/router/network/NetworkRouteSelector.java:564]: Missing ? 
if (!visitedTiles.add(tileID)) {
				continue;
			}

Comment (Dima-1) [reply OsmAnd-java/src/main/java/net/osmand/router/network/NetworkRouteSelector.java:564]: This check in the begin of next while cycle (line 531)

Comment (ivanPyrohivskyi) [reply OsmAnd-java/src/main/java/net/osmand/router/network/NetworkRouteSelector.java:564]: Agree with Dima

**Pull request merged**